modulo2/projeto-labefy/projeto-labefy/src/components/CriarPlaylist.js

```diff
@@ -0,0 +1,205 @@
+
+
+export default class CriarPlaylist extends React.Component{
+    state = {
+        inputPlaylistID: "",
+        inputMusica: "",
+        inputArtista: "",
+        inputURL: "",
+        playlist: [],
+        playlistTrack: [],
+        idPlaylist: "5"
+    }
+    
+    componentDidMount() {
+        this.getAllPlaylist()
+        this.getPlaylistTracks()
+    }
+    handleInputPlaylistID = (e) => {
+        this.setState({inputPlaylistID: e.target.value})
+    }
+    handleInputMusica = (e) => {
+        this.setState({inputMusica: e.target.value})
+    }
+    handleInputArtista = (e) => {
+        this.setState({inputArtista: e.target.value})
+    }
+    handleInputURL = (e) => {
+        this.setState({inputURL: e.target.value})
+    }
+    getAllPlaylist = () => {
+        axios.get(
+            'https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists',
+            {
+                headers: {
+                    Authorization: "edson-magnini-carver"
+                }
+            }
+        )
+        .then((res) => {
+            this.setState({ playlist: res.data.result.list })
+        })
+        .catch((err) => {
+            alert(err.response.data)
+        })
+    }
+    createPlaylist = () => {
+        const body = {
+            name: this.state.inputPlaylistID
+        }
+        axios.post(
+            'https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists', body,
+            {
+                headers: {
+                    Authorization: "edson-magnini-carver"
+                }
+            }
+        )
+        .then((res) => {
+            this.setState({inputPlaylistID: ''})
+            alert("Playlist criada")
+            this.getAllPlaylist()
+            })
+        .catch((err)=>{
+            alert(err.response.data)
+        })
+    }
+
+    deletePlaylist = (id) => {
+        axios.delete(
+            `https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists/${id}`,
+            {
+                headers: {
+                Authorization: "edson-magnini-carver"
+                }
+            }
+        )
+        .then((res) => {
+            alert("Playlist deletada")
+            this.getAllPlaylist()
+        })
+        .catch((err) => {
+            alert(err.response.data)
+        })
+    }
+
+    addTrackToPLaylist = (id) => {
+        const body = {
+            name: this.state.inputMusica,
+            artist: this.state.inputArtista,
+            url: this.state.inputURL
+        }
+        axios.post(
+            `https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists/${id}/tracks`, body,
+            {
+                headers: {
+                    Authorization: "edson-magnini-carver"
+                }
+            } 
+        )
+        .then((res) => {
+            alert("Musica adicionada")
+            this.setState({inputMusica: ''})
+            this.setState({inputArtista: ''})
+            this.setState({inputURL: ''})
```

Comment:
Nessa situação usamos o this.setState somente uma vez e alterando todos os estados. 
```
this.setState({
    inputMusica: '',
    inputArtista: '',
    inputURL: ''
})
```